### ISSUE TITLE
Document that XmlDocument.Save() creates UTF-8 files *with BOM* with an explicit "UTF-8" encoding attribute

### ISSUE THREAD

**Issue by mklement0** [dotnet/docs#10326]
https://github.com/dotnet/corefx/issues/34118 demonstrates that while `XmlDocument.Save(string)` creates _BOM-less_ UTF-8 files in the _absence_ of an `encoding` attribute, signaling UTF-8 encoding _explicitly_ via an `encoding` attribute in the XML declaration unexpectedly creates a UTF-8 file _with BOM_.

This is **problematic** for two reasons:

* From a cross-platform perspective: A document with a UTF-8 (pseudo-)BOM (Unicode signature) can cause  problems in cross-platform use, because many utilities on Unix-like platforms and, e.g., Java'ss, where many utilities neither expect nor know how to handle such  a BOM.

* From an internal-consistency perspective: UTF-8 files should be created _without BOM_, as has been the default since the inception of .NET; specifying UTF-8 explicitly should only produce a BOM _if explicitly requested_ (although the standard does allow such BOMs).
  * As an aside: a related intra-.NET inconsistency is that `System.Text.Encoding.UTF8` returns an encoding that _does_ produce a BOM, but this unexpected behavior is at least [documented](https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding.utf8?view=netframework-4.7.2#remarks)

@krwq feels that fixing this inconsistency is too much of a breaking change, so the **behavior should be documented**; **to summarize**:

When the [`XmlDocument.Save(string)`](https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmldocument.save?view=netcore-2.2#System_Xml_XmlDocument_Save_System_String_) overload is used:

* In the _absence_ of an `encoding` attribute, the `.Save(string)` method creates a UTF-8 _without BOM_, in line with .NET's default and suitable for cross-platform use.

* If a `UTF-8`-valued `encoding` attribute is present, the `.Save(string)` method creates a  UTF-8-encoded file _with BOM_.

    *  Note that it doesn't matter whether a given document was originally read from a _file_ / a _string_ with an explicit `encoding="UTF-8"` attribute (the case of `UTF-8` doesn't matter) in its XML declaration, or whether a UTF-8 `encoding` attribute was created _programmatically_ via `XmlDocument.CreateXmlDeclaration()`.

    * @krwq  demonstrates a **workaround** based on explicit creation of an `XmlWriter` instance [here](https://github.com/dotnet/corefx/issues/34118#issuecomment-454193681).
  
Finally, it's also worth mentioning that using an `encoding` value that isn't recognized (as one of the default / registered .NET encodings) causes an _exception_ on calling `.Save()` (but not on _reading_).

**Comment by krwq** [comment]
@mklement0 I agree this should be documented but it is working consistently: explicit encoding => BOM, no explicit encoding => no BOM.
Also cross-platform does not imply "must work with Linux tools" especially specification mentions that xml parsers should understand BOM

**Comment by mklement0** [comment]
> I agree this should be documented

I'm glad to hear it, and it's sufficient for us to agree on that, given the decision not to change the behavior, but ...

---

... yes, the behavior is compliant with the XML spec, but is an  _inconsistency_ nonetheless:

If I make _explicit_ what is the _default_, there should be no change in behavior - I'm opting for the _same thing_: implicitly in one case, explicitly in the other.

If the default BOM-less file format is acceptable _as a UTF-8 file_, there is no reason to deviate from that when asking for UTF-8 explicitly, unless you specifically ask for _a BOM_.

As I said, while a compliant XML parser _should_ be able to handle both no-BOM and with-BOM UTF-8 files, _pragmatically_ speaking it's not always parsers that read XML files, and a UTF-8 BOM is a perennial pain point in the Unix world.
